tests/FakeItEasy.Analyzer.VisualBasic.Tests/NonVirtualSetupAnalyzerTests.cs

```diff
@@ -0,0 +1,378 @@
+﻿namespace FakeItEasy.Analyzer.VisualBasic.Tests
+{
+    using System.Diagnostics.CodeAnalysis;
+    using FakeItEasy.Analyzer.Tests.Helpers;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Xunit;
+
+    public class NonVirtualSetupAnalyzerTests : DiagnosticVerifier
+    {
+        [Fact]
+        public void Diagnostic_Should_Not_Be_Triggered_For_Interface()
+        {
+            const string Test = @"Imports FakeItEasy
+Namespace TheNamespace
+    Class TheClass
+        Sub Test()
+            Dim foo = A.Fake(Of IFoo)()
+            A.CallTo(Function() foo.Bar()).Returns(42)
+        End Sub
+    End Class
+
+    Interface IFoo
+        Function Bar() As Integer
+    End Interface
+End Namespace
+";
+
+            this.VerifyVisualBasicDiagnostic(Test);
+        }
+
+        [Fact]
+        public void Diagnostic_Should_Not_Be_Triggered_For_Virtual_Member()

```

Comment:
> but in the user's language, we're really talking about an Overridable member. Should the test name reflect that?

Good point. I blindly copied and adapted the C# tests, but I can rename them.



> To continue the nitpickiness, capital letters and underscores seem to be competing here for the job of distinguishing words. Shouldn't we just use one of them? (I vote for capital letters.)

IMO underscores make it a little easier to read the test name as a sentence, but I don't have strong feelings either way. Whichever option we choose, I'd rather do it in a separate PR.